src/main.rs

```diff
@@ -829,9 +792,10 @@ async fn process_signed_delegate_action(
             });
         }
 
-        let actions = vec![Action::Delegate(signed_delegate_action)];
+        // FIXME: duplicate code with other branch below
```

Comment:
This will be fixed in a refactor where we have a separate `process_signed_delegate_action` fn for each config type.